options/options_helper.cc

```diff
@@ -969,6 +969,20 @@ Status OptionTypeInfo::Parse(const ConfigOptions& config_options,
       ConfigOptions copy = config_options;
       copy.invoke_prepare_options = false;
       return parse_func_(copy, opt_name, opt_value, opt_addr);
+    } else if (opt_name == "read_amp_bytes_per_bit") {
+      // A temporary hack to work around a bug in 6.10, 6.11, 6.12, 6.13 and
+      // 6.14. The bug will write out 8 bytes to OPTIONS file from the starting
+      // address of BlockBasedTableOptions.read_amp_bytes_per_bit which is
+      // actually a uint32. Consequently, the value of read_amp_bytes_per_bit
+      // written in the OPTIONS file is wrong. From 6.15, RocksDB will
+      // try to parse the read_amp_bytes_per_bit from OPTIONS file as a uint32.
+      // To be able to load OPTIONS file generated by affected releases before
+      // the fix, we need to manually parse read_amp_bytes_per_bit with this
+      // special hack.
+      assert(type_ == OptionType::kUInt32T);
+      uint64_t read_amp_bytes_per_bit = ParseUint64(opt_value);
+      EncodeFixed32(opt_addr, static_cast<uint32_t>(read_amp_bytes_per_bit));
```

Comment:
EncodeFixed32 always uses little endian. We need to respect machine order in loading the value for use (in case someone in the world uses RocksDB with big endian).

Reply:
I think it handles both little and big endian: https://github.com/facebook/rocksdb/blob/master/util/coding_lean.h#L34